### PR TITLE
[SDK/CLI] Add chat group poc

### PR DESCRIPTION
**Pull request by 0mza987**
# Description

Add chat group sdk poc:

```python
from promptflow._sdk.entities._chat_group._chat_group import ChatGroup
from promptflow._sdk.entities._chat_group._chat_role import ChatRole

copilot = ChatRole(
    flow=FLOWS_DIR / "chat_group_copilot",
    role="assistant",
    inputs=dict(
        question=topic,
        model="gpt-3.5-turbo",
        conversation_history="${parent.conversation_history}",
    ),
)
simulation = ChatRole(
    flow=FLOWS_DIR / "chat_group_simulation",
    role="user",
    inputs=dict(
        topic=topic,
        persona="criticizer",
        conversation_history="${parent.conversation_history}",
    ),
)

chat_group = ChatGroup(
    roles=[copilot, simulation],
    max_turns=4,
    max_tokens=1000,
    max_time=1000,
    stop_signal="[STOP]",
)
chat_group.invoke()

# history has 4 records
history = chat_group.conversation_history
assert len(history) == 4
assert history[0][0] == history[2][0] == copilot.role
assert history[1][0] == history[3][0] == simulation.role

```

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/_sdk/entities/_chat_group/_chat_group.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+        logger.info(
+            f"Chat group maximum turns: {max_turns!r}, maximum tokens: {max_tokens!r}, maximum time: {max_time!r}."
+        )
+        return max_turns, max_tokens, max_time
+
+    def invoke(self, *args, **kwargs):
+        """Invoke the chat group"""
+        if args or kwargs:
+            logger.warn(
+                f"Chat group invoke does not accept arguments, got {args!r} and {kwargs!r} instead and ignored them."
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/entities/_chat_group/_chat_group.py:111]: if we don't accept arguments, why are they exist in the function signature?

File: src/promptflow/promptflow/_sdk/entities/_chat_group/_chat_group.py [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+from promptflow._sdk._constants import STOP_SIGNAL, ChatGroupSpeakOrder
+from promptflow._sdk._errors import ChatGroupError
+from promptflow._sdk.entities._chat_group._chat_role import ChatRole
+from promptflow._utils.logger_utils import get_cli_sdk_logger
+
+logger = get_cli_sdk_logger()
+
+
+class ChatGroup:
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/entities/_chat_group/_chat_group.py:16]: An entity should have schema class?

File: src/promptflow/promptflow/_sdk/entities/_chat_group/__init__.py [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+# isort: skip_file
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/entities/_chat_group/__init__.py:4]: why need these skip line?

File: src/promptflow/promptflow/_sdk/entities/_chat_group/_chat_group.py [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+            logger.warn(
+                f"Chat group invoke does not accept arguments, got {args!r} and {kwargs!r} instead and ignored them."
+            )
+
+        logger.info("Invoking chat group.")
+
+        chat_round = 0
+        chat_token = 0
+        chat_start_time = time.time()
+        while True:
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/entities/_chat_group/_chat_group.py:119]: make sure we have good trace for the orchestration process

File: src/promptflow/promptflow/_sdk/entities/_chat_group/__init__.py (hunk 3 above)
Comment (0mza987) [reply src/promptflow/promptflow/_sdk/entities/_chat_group/__init__.py:4]: copied from an existing init, let me remove it.

File: src/promptflow/promptflow/_sdk/entities/_chat_group/_chat_group.py (hunk 2 above)
Comment (0mza987) [reply src/promptflow/promptflow/_sdk/entities/_chat_group/_chat_group.py:17]: I'm planning to add schema and cli experiment in the next pr

File: src/promptflow/promptflow/_sdk/entities/_chat_group/_chat_group.py (hunk 1 above)
Comment (0mza987) [reply src/promptflow/promptflow/_sdk/entities/_chat_group/_chat_group.py:119]: good point, this is the old version interface. Wil remove it.

File: src/promptflow/promptflow/_sdk/entities/_chat_group/_chat_group.py (hunk 4 above)
Comment (0mza987) [reply src/promptflow/promptflow/_sdk/entities/_chat_group/_chat_group.py:127]: Do you mean the log details? This is the current orchestration log if log level is set to INFO:
image

Comment (0mza987) [reply src/promptflow/promptflow/_sdk/entities/_chat_group/_chat_group.py:122]: Offline synced, added an item to track: 3015443

**Review by wangchao1230 — APPROVED** [review]

**Review by crazygao — APPROVED** [review]

**Pull request merged**